functions/apis/comments.js

```diff
@@ -11,62 +11,107 @@ const createComment = (req, res) => {
         updatedAt: new Date().toISOString(),
 
     };
-    if (newComment.post === undefined) {
+    if (newComment.title == undefined) {
+        return res.status(400).json({ error: "Missing required parameter" });
+    }
+    if (newComment.post == undefined) {
+        return res.status(400).json({ error: "Missing required parameter" });
+    }
+    if (newComment.username == undefined) {
         return res.status(400).json({ error: "Missing required parameter" });
     }
 
-    console.log("Starting Upload...", newComment.username);
-    db.collection(`/comments/`)
+    //TODO: validate the user publishing the post exists in the 
+    // database
+    console.log("Starting publishing comment...", newComment.username);
+    db.collection('comments')
         .add(newComment)
-        .then((doc) => {
-            return res.status(201).json({ message: "Post created successfully" });
+        .then((docRef) => {
+            return res.status(201).json({ message: "post has been published", postId: docRef.id });
         })
         .catch((error) => {
             res.status(500).json({ error: error.code });
         });
-        console.log(req);
 };
 
 
 const updatedComment = (req, res) => {
+
+    const authToken = req.headers.authorization;
+    if (!authToken) {
+        return res.status(401).json({ message: "not authenticated" });
+    }
     const comment = {
+        username: req.body.username,
+        postId: req.body.postId,
         post: req.body.post,
         title: req.body.title,
         updatedAt: new Date().toISOString(),
 
     };
-   const username = req.body.username;
 
-    if (comment.post == undefined) {
+    if (comment.postId == undefined) {

```

Comment:
if this is what the contract says it's wrong, basically the only thing we need is the commentId and the body of the comment